### PR TITLE
Integrate Sentry

### PR DESCRIPTION
**Pull request by onurtemizkan**
Ref: https://github.com/getsentry/sentry-javascript/issues/9055

Summary:

- Instrumented `entry.server`, `entry.client` and `root.tsx`.
- Integrated release / sourcemap upload to `build` script.
- Set `customerAccessToken` as a Sentry tag on server-side.
- Added pages to trigger different types of errors and navigation transactions.

### CODE REVIEW

File: app/root.tsx [hunk 1]
```diff
@@ -60,6 +62,12 @@ export function links() {
 export async function loader({context}: LoaderArgs) {
   const {storefront, session, cart} = context;
   const customerAccessToken = await session.get('customerAccessToken');
+
+  // Set customerAccessToken as a Sentry tag
+  Sentry.configureScope(scope => {
+    scope.setTag('customerAccessToken', customerAccessToken?.accessToken);
```
Comment (AbhiPrasad) [inline app/root.tsx]: Let's not add this as a tag, from what I see in the docs it works like a session token, so I don't think it's huge value to users.

File: README.md [hunk 2]
```diff
@@ -1,30 +1,44 @@
...
+
+```ts
+Sentry.init({
+  // TODO: replace with your Sentry DSN
+  dsn: 'SENTRY_DSN',
+  // ...
+});
+```
+3. Place your Sentry Organization Slug, Project Slug, and Auth Token in `.env`
+This is required to create Sentry releases and upload source maps.
```
Comment (AbhiPrasad) [inline README.md:31]: Maybe we can link to docs for this? https://docs.sentry.io/platforms/javascript/guides/remix/sourcemaps/#2-create-a-release-and-upload-source-maps

File: README.md [hunk 3]
```diff
@@ -1,30 +1,44 @@
...
 ## Getting started
 
 **Requirements:**
 
 - Node.js version 16.14.0 or higher
 
+**Setup:**
+
+1. `npm install`
+2. Place your DSN in `entry.client.tsx` and `entry.server.tsx`
```
Comment (AbhiPrasad) [inline README.md:21]: Should we populate DSN from environmental variable as well?

**Review by AbhiPrasad — COMMENTED** [review]
cool!

File: README.md (hunk 3 above)
Comment (onurtemizkan) [reply README.md:21]: Just tried it, and interestingly it seems the env variables in Hydrogen apps are not directly accessible from `process` (which is not defined). They are accessible inside loaders, actions and request handlers as in `context` objects. We can't access them directly from the root of `entry.server.tsx`. https://github.com/Shopify/hydrogen/issues/714

There's probably a workaround but IMHO including that might make this demo look more complicated than it is.

Comment (AbhiPrasad) [reply README.md:21]: alright, let's not worry about it then.

**Review by AbhiPrasad — APPROVED** [review]

**Pull request merged**